server/src-lib/Hasura/GraphQL/Resolve/Select.hs

```diff
@@ -44,6 +45,27 @@ withSelSet selSet f =
     res <- f fld
     return (G.unName $ G.unAlias $ _fAlias fld, res)
 
+jsonPathToColExp :: (MonadError QErr m) => T.Text -> m S.SQLExp
+jsonPathToColExp t = case parseJSONPath t of
+  Left s       -> throwVE $ T.pack s
```

Comment:
`throwVE` gives `validation-failed` exception. It is thrown only in case of GraphQL schema validation errors. Here it is parsing error. 

Use `throw400 ParseFailed $ "json path parse error: " <> T.pack s`

You're free to use any convenient error message other than above.